pubsub/mqtt/mqtt.go

```diff
@@ -213,8 +209,9 @@ func (m *mqttPubSub) Subscribe(req pubsub.SubscribeRequest, handler pubsub.Handl
 	m.consumer = c
 
 	go func() {
-		token := m.consumer.SubscribeMultiple(
```

Comment:
I think you need to keep SubscribeMultiple. The connection is re-created every time the `Subscribe` method is invoked, so you need to re-register all topic subscriptions. Otherwise only the last one will work.

Reply:
@daixiang0 please add a test to verify this works as intended.

Reply:
It is what I said before (`now it does not support subscribing two topics with different callback`).

@ItalyPaleAle I check SubscribeMultiple API, it registers all topic to one subscribation, so only the last one will work:
```
	// SubscribeMultiple starts a new subscription for multiple topics. Provide a MessageHandler to
	// be executed when a message is published on one of the topics provided, or nil for the
	// default handler.
	//
	// If options.OrderMatters is true (the default) then callback must not block or
	// call functions within this package that may block (e.g. Publish) other than in
	// a new go routine.
	// callback must be safe for concurrent use by multiple goroutines.
	SubscribeMultiple(filters map[string]byte, callback MessageHandler) Token
```

Reply:
Thanks for confirming!

Please take a look at #1737 which does support multiple handlers.

Reply:
Oh I miss it, thank you!